plugins/outputs/remotefile/remotefile.go

```diff
@@ -0,0 +1,239 @@
+	FinalWriteTimeout config.Duration `toml:"final_write_timeout"`
+	WriteBackInterval config.Duration `toml:"cache_write_back"`
+	MaxCacheSize      config.Size     `toml:"cache_max_size"`
+	UseBatchFormat    bool            `toml:"use_batch_format"`
+	Trace             bool            `toml:"trace"`
+	Log               telegraf.Logger `toml:"-"`
+
+	root     *vfs.VFS
+	fscancel context.CancelFunc
+	vfsopts  vfscommon.Options
+
+	templates  []*template.Template
+	serializer serializers.Serializer
+}
+
+func (*File) SampleConfig() string {
+	return sampleConfig
+}
+
+func (f *File) SetSerializer(serializer serializers.Serializer) {
+	f.serializer = serializer
+}
+
+func (f *File) Init() error {
+	if len(f.Files) == 0 {
+		return errors.New("no files specified")
+	}
+
+	// Set defaults
+	if f.Remote.Empty() {
+		f.Remote.Set([]byte("local"))
+	}
+
+	if f.FinalWriteTimeout <= 0 {
+		f.FinalWriteTimeout = config.Duration(10 * time.Second)
+	}
+
+	// Prepare VFS options
+	f.vfsopts = vfscommon.DefaultOpt
+	f.vfsopts.CacheMode = vfscommon.CacheModeWrites // required for appends
+	if f.WriteBackInterval > 0 {
+		f.vfsopts.WriteBack = time.Duration(f.WriteBackInterval)
+	}
+	if f.MaxCacheSize > 0 {
+		f.vfsopts.CacheMaxSize = fs.SizeSuffix(f.MaxCacheSize)
+	}
+
+	// Redirect logging
+	fs.LogPrint = func(level fs.LogLevel, text string) {
+		if !f.Trace {
+			return
+		}
+		f.Log.Debugf("[%s] %s", level.String(), text)
+	}
+
+	// Setup custom template functions
+	funcs := template.FuncMap{"now": time.Now}
+
+	// Setup filename templates
+	f.templates = make([]*template.Template, 0, len(f.Files))
+	for _, ftmpl := range f.Files {
+		tmpl, err := template.New(ftmpl).Funcs(funcs).Parse(ftmpl)
+		if err != nil {
+			return fmt.Errorf("parsing file template %q failed: %w", ftmpl, err)
+		}
+		f.templates = append(f.templates, tmpl)
+	}
+
+	return nil
+}
+
+func (f *File) Connect() error {
+	remoteRaw, err := f.Remote.Get()
+	if err != nil {
+		return fmt.Errorf("getting remote secret failed: %w", err)
+	}
+	remote := remoteRaw.String()
+	remoteRaw.Destroy()
+
+	// Construct the underlying filesystem config
+	parsed, err := fspath.Parse(remote)
+	if err != nil {
+		return fmt.Errorf("parsing remote failed: %w", err)
+	}
+	info, err := fs.Find(parsed.Name)
+	if err != nil {
+		return fmt.Errorf("cannot find remote type %q: %w", parsed.Name, err)
+	}
+
+	// Setup the remote virtual filesystem
+	ctx, cancel := context.WithCancel(context.Background())
+	rootfs, err := info.NewFs(ctx, parsed.Name, parsed.Path, fs.ConfigMap(info, parsed.Name, parsed.Config))
+	if err != nil {
+		cancel()
+		return fmt.Errorf("creating remote failed: %w", err)
+	}
+	f.fscancel = cancel
+	f.root = vfs.New(rootfs, &f.vfsopts)
+
+	// Force connection to make sure we actually can connect
+	if _, err := f.root.Fs().List(ctx, "/"); err != nil {
+		return err
+	}
+	total, used, free := f.root.Statfs()
+	f.Log.Debugf("Connected to %s with %s total, %s used and %s free!",
```

Comment:
What does this show for something like S3 where you don't really have a limit?

Reply:
Hi @powersj, I tested this build, for S3 is says (for a bucket that has some data in it):
`D! [outputs.remotefile]  Connected to S3 bucket telegrafofficial with 1.1 PB total, 0 B used and 1.1 PB free!
`

Reply:
thanks for sharing that! I think this is fine. My fear is it showed something like 0 for total.